src/components/calcite-tooltip/calcite-tooltip.tsx

```diff
@@ -11,6 +11,207 @@ import {
   OverlayPositioning
 } from "../../utils/popper";
 import { queryElementRoots } from "../../utils/dom";
+import { getKey } from "../../utils/key";
+
+class TooltipManagerSingleton {
+  keyUpHandler = (event: KeyboardEvent): void => {
+    if (getKey(event.key) === "Escape") {
+      const tooltipEl = this.registeredElements.get(event.currentTarget as HTMLElement);
+
+      if (tooltipEl) {
+        this.clearHoverTimeout(tooltipEl);
+        this.toggleTooltip(tooltipEl, false);
+      }
+    }
+  };
+
+  mouseEnterShow = (event: MouseEvent): void => {
+    this.hoverEvent(event, true);
+  };
+
+  mouseLeaveHide = (event: MouseEvent): void => {
+    this.hoverEvent(event, false);
+  };
+
+  clickHandler = (event: MouseEvent): void => {
+    const clickedTooltip = this.registeredElements.get(event.currentTarget as HTMLElement);
+
+    this.clickedTooltip = clickedTooltip;
+
+    if (clickedTooltip) {
+      this.toggleTooltip(clickedTooltip, false);
+    }
+  };
+
+  focusShow = (event: FocusEvent): void => {
+    this.focusEvent(event, true);
+  };
+
+  blurHide = (event: FocusEvent): void => {
+    this.focusEvent(event, false);
+  };
+
+  hoverTimeouts: WeakMap<HTMLCalciteTooltipElement, number> = new WeakMap();
+
+  clickedTooltip: HTMLCalciteTooltipElement;
+
+  activeTooltipEl: HTMLCalciteTooltipElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * CSS Selector to match reference elements for tooltips. Reference elements will be identified by this selector in order to open their associated tooltip.
+   * @default `[data-calcite-tooltip-reference]`
+   */
+  selector = `[${TOOLTIP_REFERENCE}]`;
+
+  registeredElements = new WeakMap<HTMLElement, HTMLCalciteTooltipElement>();
+
+  registerElement(element: HTMLElement, tooltip: HTMLCalciteTooltipElement): void {
+    this.registeredElements.set(element, tooltip);
+
+    element.addEventListener("keyup", this.keyUpHandler);
+    element.addEventListener("mouseover", this.mouseEnterShow);
+    element.addEventListener("mouseout", this.mouseLeaveHide);
+    element.addEventListener("click", this.clickHandler);
+    element.addEventListener("focus", this.focusShow);
+    element.addEventListener("blur", this.blurHide);
```

Comment:
The only downside i see is that every tooltip needs all these events. The reason for the manager, was to have delegated events so that the manager only has these events and not every tooltip instance.